File: seminar_2/task_3/task3.cpp
#include <iostream>
#include <cmath>
using namespace std;

int integer_square(int n){
    int l = 0, r = n;
    while (l <= r){
        int mid = (l+r) / 2;
        if(mid * mid > n)
            r = mid-1;
        else if(mid*mid < n)
            l = mid+1;
        else
            return mid;}
    return r;
}

int main(void){
    int mistakes = 0;
    for(int i=0; i<=10000; i++)
        mistakes+=integer_square(i)!=int(sqrt(i));
    cout<<mistakes;
    return 0;
}

File: seminar_2/task_3/task_3.py

# третья задачка
# найти корнеь числа (ближайшее целое)

def sqrt_nearest(target):
    l = 0
    r = target 
    
    while l <= r:
        
        mid = (l + r) // 2
        
        if mid * mid == target:
            return mid
        elif mid * mid > target:
            r = mid - 1
        else:
            l = mid + 1
        
        print(l, r)
    return mid

print(sqrt_nearest(63))

